app/src/main/java/com/courseproject/tindar/usecases/login/LoginInteractor.java

```diff
@@ -0,0 +1,20 @@
+package com.courseproject.tindar.usecases.login;
+
+public class LoginInteractor implements LoginInputBoundary {
+
+    final LoginDsGateway loginDsGateway;
+
+    public LoginInteractor(LoginDsGateway loginDsGateway){
+        this.loginDsGateway = loginDsGateway;
+    }
+
+    @Override
+    public Boolean checkUserPassword(String email, String password){
+        return !(loginDsGateway.readUserId(email, password) == null);
+    }
```

Comment:
I think instead of having this function you can just check in the frontend activity if !(loginDsGateway.readUserId(email, password) == null)) {...} else {...}

In that way you have one less method to test.

Reply:
:D